### PR TITLE
Fixed two MR bugs:

### PR DESCRIPTION
**Pull request by chtyim**
- In MapReduceContextProvider, we should always get the MapReduceClassLoader from hConf, not Confer
- Do proper resource cleanup for the artifact plugin instantiator

### CODE REVIEW

**Comment by gokulavasan** [comment]
LGTM :+1:

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java [hunk 1]
```diff
@@ -249,6 +251,7 @@ public void starting() {
 
       @Override
       public void terminated(Service.State from) {
+        Closeables.closeQuietly(artifactPluginInstantiator);
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java]: @chtyim Looks like we also close it in two more places in this class.

**Comment by gokulavasan** [comment]
One comment. LGTM otherwise!

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java (hunk 1 above)
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java]: Good catch.

**Comment by gokulavasan** [comment]
LGTM :+1:

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT2644-3

**Pull request merged**